reporting-app/app/models/ex_parte_activity.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# ExParteActivity stores hours data submitted via API or batch upload.
+#
+# These are "ex parte" (automated/external) hours as opposed to manually
+# reported hours from ActivityReportApplicationForm. Ex parte hours are
+# auto-verified and don't require staff review.
+#
+# @note The `outside_period` flag should be set by the service layer
+#   (ExParteActivityService) when creating entries, not by callbacks.
+#
+class ExParteActivity < ApplicationRecord
+  ALLOWED_CATEGORIES = %w[employment community_service education].freeze
+
+  SOURCE_TYPE_API = "api"
+  SOURCE_TYPE_BATCH = "batch_upload"
+  ALLOWED_SOURCE_TYPES = [ SOURCE_TYPE_API, SOURCE_TYPE_BATCH ].freeze
+
+  MAX_HOURS = 744
+
+  # --- Associations ---
+
+  belongs_to :certification, optional: true
+
+  # --- Validations ---
+
+  validates :member_id, presence: true
+  validates :category, presence: true, inclusion: { in: ALLOWED_CATEGORIES }
+  validates :hours, presence: true,
+                    numericality: { greater_than: 0, less_than_or_equal_to: MAX_HOURS }
+  validates :period_start, presence: true
+  validates :period_end, presence: true
```

Comment:
I think you can use SDK's DateRange type

Reply:
I thought of that but since we are likely not using this table on a form and don't need complex valiudation it might be overkill here our three line validation does all we need :
`validates :period_start, presence: true
   validates :period_end, presence: true
   validate :period_end_after_start`

Reply:
I will change for future needs.